File: BioStructures/parse_mmtf.jl
# Benchmark the parsing of a MMTF file given as an argument

using BioStructures

mmtf_filepath = ARGS[1]

# Run to JIT compile
read(mmtf_filepath, MMTF)

elapsed = @elapsed struc = read(mmtf_filepath, MMTF)

println(elapsed)

File: biotite/parse_mmtf.py
# Benchmark the parsing of a MMTF file given as an argument

import sys
import time
import biotite.structure.io.mmtf as mmtf

mmtf_filepath = sys.argv[1]

start = time.time()
file = mmtf.MMTFFile()
file.read(mmtf_filepath)
mmtf.get_structure(file)
end = time.time()

print(end - start)
